src/shared-minter-filter-mapping.ts

```diff
@@ -0,0 +1,335 @@
+  if (!contractAllowlist.minterContractAllowlist.includes(minter.id)) {
+    contractAllowlist.minterContractAllowlist = contractAllowlist.minterContractAllowlist.concat(
+      [minter.id]
+    );
+    contractAllowlist.updatedAt = event.block.timestamp;
+    contractAllowlist.save();
+  }
+}
+
+export function handleMinterRevokedForContract(
+  event: MinterRevokedForContract
+): void {
+  let minterFilter = loadOrCreateSharedMinterFilter(
+    event.address,
+    event.block.timestamp
+  );
+
+  let minter = loadOrCreateMinter(event.params.minter, event.block.timestamp);
+
+  // log a warning if the minter's minter filter does not match the minter
+  // filter that emitted the MinterRevokedForContract event
+  if (minter.minterFilter != minterFilter.id) {
+    log.warning(
+      "[WARN] Contract allowlisted minter at {} does not match minter filter that emitted the MinterRevokedForContract event at {}",
+      [minter.minterFilter, minterFilter.id]
+    );
+  }
+
+  // load or create the contract-specific allowlist for the MinterFilter
+  let contractAllowlist = loadOrCreateMinterFilterContractAllowlist(
+    minterFilter,
+    event.params.coreContract,
+    event.block.timestamp
+  );
+
+  // remove minter from the list of contract allowlisted minters
+  let newMinterContractAllowlist: string[] = [];
+  for (let i = 0; i < contractAllowlist.minterContractAllowlist.length; i++) {
+    if (contractAllowlist.minterContractAllowlist[i] != minter.id) {
+      newMinterContractAllowlist.push(
+        contractAllowlist.minterContractAllowlist[i]
+      );
+    }
+  }
+  contractAllowlist.minterContractAllowlist = newMinterContractAllowlist;
+  contractAllowlist.updatedAt = event.block.timestamp;
+  contractAllowlist.save();
+
+  // if the contract allowlist is empty, delete the entity
+  if (contractAllowlist.minterContractAllowlist.length == 0) {
+    store.remove("MinterFilterContractAllowlist", contractAllowlist.id);
+  }
+}
+
+export function handleProjectMinterRegistered(
+  event: ProjectMinterRegistered
+): void {
+  let minterFilter = loadOrCreateSharedMinterFilter(
+    event.address,
+    event.block.timestamp
+  );
+
+  let minter = loadOrCreateMinter(event.params.minter, event.block.timestamp);
+
+  // log a warning if the minter's minter filter does not match the minter
+  // filter that emitted the ProjectMinterRegistered event
+  if (minter.minterFilter != minterFilter.id) {
+    log.warning(
+      "[WARN] Project minter at {} does not match minter filter that emitted the ProjectMinterRegistered event at {}",
+      [minter.minterFilter, minterFilter.id]
+    );
+  }
+
+  // if the project's minter filter does not match the minter filter that
+  // emitted the ProjectMinterRemoved event, then this is a pre-configuring
+  // event, and we should return early and not update the project's minter
+  const coreContract = Contract.load(event.params.coreContract.toHexString());
```

Comment:
Nit: Comments definitely make this clear, but might be slightly easier to read at first glance if we assign coreContract.minterFilter to a descriptive variable like currentProjectMinterFilter or something

Reply:
I've updated this section of code to be more readable, using the variable name `currentMinterFilter` 👍